packages/spotlight/src/vite-plugin.ts

```diff
@@ -0,0 +1,260 @@
+ * https://github.com/getsentry/sentry/blob/master/src/sentry/lang/javascript/processor.py#L67
+ *
+ * @param str An object that contains serializable values
+ * @param max Maximum number of characters in truncated string
+ * @returns string Encoded
+ */
+function snipLine(line: string, colno: number): string {
+  let newLine = line;
+  const lineLength = newLine.length;
+  if (lineLength <= 150) {
+    return newLine;
+  }
+  if (colno > lineLength) {
+    // eslint-disable-next-line no-param-reassign
+    colno = lineLength;
+  }
+
+  let start = Math.max(colno - 60, 0);
+  if (start < 5) {
+    start = 0;
+  }
+
+  let end = Math.min(start + 140, lineLength);
+  if (end > lineLength - 5) {
+    end = lineLength;
+  }
+  if (end === lineLength) {
+    start = Math.max(end - 140, 0);
+  }
+
+  newLine = newLine.slice(start, end);
+  if (start > 0) {
+    newLine = `'{snip} ${newLine}`;
+  }
+  if (end < lineLength) {
+    newLine += ' {snip}';
+  }
+
+  return newLine;
+}
+
+function isValidSentryStackFrame(frame: SentryStackFrame): frame is ValidSentryStackFrame {
+  return !!frame.filename && !!frame.lineno && !!frame.colno;
+}
+
+export const CONTEXT_LINES_ENDPOINT = '/spotlight/contextlines';
+export const sourceContextMiddleware: Connect.NextHandleFunction = function (req, res, next) {
+  // We're only interested in handling a PUT request to CONTEXT_LINES_ENDPOINT
+  if (req.url !== CONTEXT_LINES_ENDPOINT || req.method !== 'PUT') {
+    return next();
+  }
+
+  let requestBody = '';
+  req.on('data', chunk => {
+    requestBody += chunk;
+  });
+
+  req.on('end', async () => {
+    const stacktrace = parseStackTrace(requestBody);
+
+    if (!stacktrace) {
+      res.writeHead(500);
+      res.end();
+      return;
+    }
+
+    for (const frame of stacktrace.frames ?? []) {
+      if (
+        isValidSentryStackFrame(frame) &&
+        // let's ignore dependencies for now with this naive check
+        !frame.filename.includes('/node_modules/')
+      ) {
+        const generatedCode = await getGeneratedCodeFromServer(frame.filename);
+        if (!generatedCode) {
+          continue;
+        }
+
+        // Extract the inline source map from the minified code
+        const inlineSourceMapMatch = generatedCode.match(/\/\/# sourceMappingURL=data:application\/json;base64,(.*)/);
+
+        if (inlineSourceMapMatch && inlineSourceMapMatch[1]) {
+          const sourceMapBase64 = inlineSourceMapMatch[1];
+          const sourceMapContent = Buffer.from(sourceMapBase64, 'base64').toString('utf-8');
+          await applySourceContextToFrame(sourceMapContent, frame);
+        }
+      }
+    }
+
+    const responseJson = JSON.stringify(stacktrace);
+
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+    res.end(responseJson);
+  });
+};
+
+export default function spotlight({ port }: { port?: number } = {}): PluginOption {
+  Sentry.init({
+    tracesSampleRate: 0,
+    spotlight: true,
+  });
```

Comment:
Why do we init the SDK here? Shouldn't spotlight be decoupled from the SDK (might have missed some decisions in the last weeks)? Most spotlight users will use the `@sentry/astro` SDK anyway

Reply:
This is on the compiler side and since this is a generic `vite` plugin, we are not guaranteed to have a Sentry SDK, especially on the compiler suite. The main reason is to be able to relay the compiler errors to Spotlight for now. I think I can hack something together to remove the need for a full SDK tbh.

Reply:
Done, thanks a lot for the push again :)

Reply:
thanks for resolving this!